homeassistant/components/danfoss_air.py

```diff
@@ -0,0 +1,89 @@
+"""
+Support for Danfoss Air HRV.
+
+Configuration:
+    danfoss_air:
+        host: IP_OF_CCM_MODULE
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/danfoss_air/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['pydanfossair==0.0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+DANFOSS_AIR_PLATFORMS = ['sensor', 'binary_sensor']
+DOMAIN = 'danfoss_air'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Danfoss Air component."""
+    conf = config[DOMAIN]
+
+    danfoss_config = {}
+    danfoss_config['host'] = conf[CONF_HOST]
+
+    hass.data['DANFOSS_DO'] = DanfossAir(danfoss_config)
+
+    for platform in DANFOSS_AIR_PLATFORMS:
+        discovery.load_platform(hass, platform, DOMAIN, {}, config)
+
+    return True
+
+
+class DanfossAir:
+    """Handle all communication with Danfoss Air CCM unit."""
+
+    def __init__(self, host):
+        """Initialize the Danfoss Air CCM connection."""
+        self._data = {}
+
+        from pydanfossair.danfossclient import DanfossClient
+        self._client = DanfossClient(host)
+
+    def get_value(self, item):
+        """Get value for sensor."""
+        if item in self._data:
+            return self._data[item]
+
+        return None
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Use the data from Digital Ocean API."""
+        _LOGGER.debug("Fetching data from Danfoss Air CCM module")
+        from pydanfossair.commands import ReadCommand
+        self._data["EXHAUST_TEMPERATURE"] \
```

Comment:
Use a constant for each of the pieces that you are storing in `self._data`. That makes it less brittle to access them from the platforms.

Alternatively, I'd recommend storing each of these attributes in their own instance variables and accessing each of them via getters on the class instead.

Reply:
Would it be OK to use the enum, that is used as argument to the command method, as constant? Reason is that a bit down the road the DanfossClient will be extended with a read_all method that returns all the values in a dictionary with the enum as key and I can the use that all the places.

Example: 
self._data[ReadCommand.exhaustTemperature] \
            = self._client.command(ReadCommand.exhaustTemperature)